src/extractor/guidance/intersection_handler.cpp

```diff
@@ -68,6 +68,17 @@ TurnType::Enum IntersectionHandler::findBasicTurnType(const EdgeID via_edge,
     return TurnType::Turn;
 }
 
+TurnInstruction IntersectionHandler::notificationOnModeChange(const TravelMode in_mode,
+                                                              const TravelMode out_mode,
+                                                              const double turnAngle,
```

Comment:
passing the turn angle in here does not really provide any benefit. The entire function is only a mimic of the basic `?` operator. I would prefer using `{in_mode == out_mode ? TurnType::Suppressed : TurnType::Notification, getTurnDirection(turnAngle)}` instead of having a dedicated function for it.

It's a single line, which can be grasped instantly, and on the other side a function that I have to search for first.

If you prefer to use the function, the function should return a TurnType and not care about the turn-angle.

So we would get `return {notificationOnModeChange(in_mode,out_mode,TurnType::Suppressed),getTurnDirection(turnAngle)}`.

Passing the turn-angle in adds complexity to the function using something that is not even considered except for the return.